### PR TITLE
##150967786 Refactors code to implement flux architecture 

### PR DESCRIPTION
**Pull request by jmnyarega**
**What the PR does**
- Implements a flux architecture using react redux
- Implements pagination 
- Disables flickering of the items when navigating through pagination links
- Add solid color on the card images
- organises project folders
- Changes drawbar from the right side of the screen to the left side of the screen
- Creates tests
- improves responsiveness of the application

**How this can be tested manually**
- go to `https://bucketlistfrontend.herokuapp.com/`
- create an account
- login in to the account
- test the above features

**Pivotal tracker refrences**
- ##150967786

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,102 +1,43 @@
 {
-  "name": "bucketlist",
+  "name": "redux-react-apps",
```
Comment (brayoh) [inline package.json]: Change this to the project name

File: package.json [hunk 2]
```diff
@@ -1,102 +1,43 @@
...
-    ]
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "eject": "react-scripts eject",
+    "test": "react-scripts test --env=jsdom --coverage"
   },
-  "eslintConfig": {
-    "extends": "react-app"
-  }
+  "description": "This project was bootstrapped with [Create React App](https://github.com/facebookincubator/create-react-app).",
```
Comment (brayoh) [inline package.json]: Change the description to point refer to the bucketlist project

File: src/__tests__/reducers.test.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+describe('Login Reducer', () => {
+it('has a default state', () => {
+  expect(loginReducer(undefined, {type:'unexpected'})).toEqual({
+    message: ''
+  });
+});
+it('can handle request ', () => {
+  expect(loginReducer("",{type:'LOGIN_USER', message: 'Hi'})).toEqual({message: "Hi"});
+});
+
```
Comment (brayoh) [inline src/__tests__/reducers.test.js]: indent your code for better readability

File: src/actions/bucketlistActions.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
+import $ from 'jquery';
```
Comment (brayoh) [inline src/actions/bucketlistActions.js]: Using jquery in react is considered a bad practice due to the conflict in the way react handles dom activities, use axios or the fetch api to make your api calls

File: src/dist/css/custom.css [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+  .large-Grid--guttersLg > .Grid-cell {
+    padding: 1.5em 0 0 1.5em;
+  }
+  .large-Grid--guttersXl {
+    margin: -2em 0 2em -2em;
+  }
+  .large-Grid--guttersXl > .Grid-cell {
+    padding: 2em 0 0 2em;
+  }
+}
```
Comment (brayoh) [inline src/dist/css/custom.css]: CSS class names should be in lowercase

File: src/common_pages/navbar_user.js [hunk 6]
```diff
@@ -1,44 +0,0 @@
...
-  }
-
-  handleToggle = () => this.setState({open: !this.state.open});
-
-  logout(){
-    localStorage.clear();
-    window.location.replace('/');
-  }
-  bucketLink(){
-    window.location.replace('/buckets/');
```
[the remark this answers is not in the document]
Comment (jmnyarega) [reply src/common_pages/navbar_user.js]: @kimani-ndegwa  and @brayoh am working on that  to ensure am  using `fetch` instead of `jquery`

**Pull request merged**